### PR TITLE
Cable management improvements

### PR DESCRIPTION
**Pull request by cedric-anne**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #10720

### CODE REVIEW

**Comment by stonebuzz** [comment]
1. Make it possible to add multiple sockets for an asset
Ok for this improvment.

IMHO :
you have to hide the "connected to" part which is useless for bulk creation
 
image

```From```, ```To```, ```Prefix``` and ```Suffix``` are fine.

But ```Position``` must be incremented

image


2. Split up the network port form, because it is confusing when using passive devices

Agree with that (This indicates that the link is made to the asset and possibly to the network port (if available))
but be careful :

when ```itemtype``` /  ```items_id``` is changed the 'Network Port' label disappears

image

3. Make it easier to add/edit connected cables to a socket 

After adding a socket to a cable, i have an error from socket form (previously linked)

image

4. Remove "wiring_side" from Socket and put it to Cable

Some customers want to finely manage sockets (rear and front)

That's why we want to keep the information of the wiring side in the socket (it is placed in front or rear)

image

 Perhaps we can also have this information in the ```Cable``` in addition to the ```Socket``` 

Such as the ```position``` of the ```NetworkPort``` and the associated ```Socket``` (when a ```Socket``` is created, if a ```NetworkPort``` is defined , ```Socket``` take the same ```position```  )

**Comment by cedric-anne** [comment]
ping @rbrannath

**Comment by rbrannath** [comment]
Done some work:

- position is increment in bulk mode (add several sockets)
- position is hidden in bulk mode
- network port is hidden in bulk mode
- restored wiring_side in sockets
- fixed disappearing network port label

**Comment by rbrannath** [comment]
> ping @rbrannath

@cedric-anne, can you help me with the failed checks? I am unaware of the concept of changing database tables between versions.

**Comment by cedric-anne** [comment]
> > ping @rbrannath
> 
> @cedric-anne, can you help me with the failed checks? I am unaware of the concept of changing database tables between versions.

Hi,

First, could you use the rebase feature of Git to have a clean pull request (i.e. only containing your commits/changes) ? You can do it using `git pull --rebase git@github.com:glpi-project/glpi.git main` command.

**Comment by rbrannath** [comment]
@cedric-anne this should be fine now

**Comment by stonebuzz** [comment]
@rbrannath have you see my changes requests ?

Best regards

**Comment by KongNan** [comment]
The pr is still in draft status though  :(   Sorry to comment here, but of all the pr's in 10.1, this is the one I'm most concerned about, and I've even thought about skipping the 10.0 stable release and updating to it directly...  Hope it's still progressing 👍

**Comment by rbrannath** [comment]
> @rbrannath have you see my changes requests ?
> 
> Best regards

Hi @stonebuzz,

no, I haven't seen it. Where do I have to look?

**Comment by stonebuzz** [comment]
> > @rbrannath have you see my changes requests ?
> > Best regards
> 
> Hi @stonebuzz,
> 
> no, I haven't seen it. Where do I have to look?

here, in the discussion thread, you will find the requests for changes

image

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Comment by stonebuzz** [comment]
> > > @rbrannath have you see my changes requests ?
> > > Best regards
> > 
> > 
> > Hi @stonebuzz,
> > no, I haven't seen it. Where do I have to look?
> 
> here, in the discussion thread, you will find the requests for changes
> 
> image

Sorry my review was in pending state

**Comment by KongNan** [comment]
all check passed! congratulations and thanks to rbrannath ✿✪❉✼♡

**Comment by KongNan** [comment]
It looks like it is now time for 10.0.1 . But after rbrannath's excellent efforts, this pr is still in draft status, which worries me. Even though we may have to wait for 10.1.0 to officially have this feature, I wonder if this pr will be approved/merged as soon as possible, just like the [other pr's in 10.1.0](https://github.com/glpi-project/glpi/milestone/49?closed=1)?

(Sorry for my bad English, and I'm not sure if my statement is appropriate, so please let me know if I'm doing it wrong)

**Comment by KongNan** [comment]
10.0.1 has already released.  now please consider pay attention to this great feature, so everyone will get benefit ~  @stonebuzz 
(sorry for my English)

File: src/Socket.php [hunk 1]
```diff
@@ -197,6 +213,93 @@ public function showForm($ID, array $options = [])
...
+                        WHEN "PassiveDCEquipment" THEN glpi_passivedcequipments_b.name
+                        ELSE NULL
+                    END endpoint_b_name'
+                ),
+                'FROM'   => 'glpi_cables',
+                'LEFT JOIN' => $leftJoins
+            ];
+
+            $rearCable = $DB->request(array_merge($criteria, $where(1, $ID)))->current();
+            $frontCable = $DB->request(array_merge($criteria, $where(2, $ID)))->current();
```
Comment (stonebuzz) [inline src/Socket.php:302]: ```$rearCable``` and ```$frontCable``` are never used
```global $DB;``` are never called

File: templates/pages/assets/socket.html.twig [hunk 2]
```diff
@@ -45,33 +45,196 @@
...
+	      wiring_side,
+	      __('Wiring side'),
+	   ) }}
+
+	{% if params["several"] is not defined %}
+		{{ fields.numberField(
+         'position',
+         item.fields['position'],
+         __('Position')
+      ) }}
```
Comment (stonebuzz) [inline templates/pages/assets/socket.html.twig:97]: ```Position``` and ```NetworkPort``` field are duplicated from creater form

image

After creation we have a blank page


```shell
[2022-06-14 09:28:49] glpiphplog.WARNING: Test logger  
[2022-06-14 09:28:49] glpiphplog.WARNING: Test logger  
[2022-06-14 09:49:51] glpiphplog.CRITICAL:   *** Uncaught Exception Error: Call to a member function request() on null in /home/stanislas/Teclib/dev/GLPI/cable_management/src/Socket.php at line 301
  Backtrace :
  src/CommonGLPI.php:655                             Glpi\Socket->showForm()
  ajax/common.tabs.php:107                           CommonGLPI::displayStandardTab()
```
image

File: templates/pages/assets/socket.html.twig [hunk 3]
```diff
@@ -45,33 +45,196 @@
       __('Position')
    ) }}
 
+	{% if params["several"] is defined %}
```
Comment (stonebuzz) [inline templates/pages/assets/socket.html.twig:48]: ```several``` form does not seem to work

image

image

**Review by stonebuzz — CHANGES_REQUESTED** [review]

**Comment by stonebuzz** [comment]
@rbrannath can you try this PR -> https://github.com/glpi-project/glpi/pull/11893

This PR allows you to add multiple sockets from one item

**Comment by KongNan** [comment]
dear @rbrannath <https://github.com/rbrannath> , could you please have a
look at stonebuzz's message? He/she replied your 11186, and make another
two #11893 <https://github.com/glpi-project/glpi/pull/11893> and #11895
<https://github.com/glpi-project/glpi/pull/11895>. Hope I can have a
PatchPanel with all of them in the coming 10.1.0 :D


Stanislas ***@***.***> 于2022年6月14日周二 17:42写道：

**Comment by rbrannath** [comment]
> @rbrannath can you try this PR -> #11893
> 
> This PR allows you to add multiple sockets from one item

@stonebuzz, sorry for the late reply, I was busy at work and then on vacation.
I tested it and I will write my results and thoughts inside that PR

**Comment by cedric-anne** [comment]
I close this PR as it seems stale and replaced by #11893  and #11895. Feel free to reopen your own PR.

**Pull request closed without merging**